### ISSUE TITLE
pydantic __init_sublcass__ not validated

### ISSUE THREAD

**Issue by Fogapod** [astral-sh/ty#3585]
### Summary

Both basedpyright and ty miss this entirely. This code produces no error, class variable type is unknown and unchecked. `__init_subclass__` class variables are validated without pydantic.

```python
from typing import Any

from pydantic import BaseModel

class Foo(BaseModel):
    def __init_subclass__(cls, *, doctype: str, **kwargs: Any):
        super().__init_subclass__(**kwargs)


class Bar(Foo):  # no error
    pass

class Bar2(Foo, doctype=123):  # no error
    pass
```

### Version

ty 0.0.40

**Comment by Fogapod** [comment]
I must've used outdated version of ty at some point. This works now. Both missing argument and wrong type are caught
